src/components/home/Home.js

```diff
@@ -23,6 +25,12 @@ function Home() {
         api: null,
     });
 
+    useEffect(() => {
+        if (!user.id) {
+            navigate("/profile");
+        }
+    });
```

Comment:
don't forget the (empty) dependency array here.

If there is no dependency array, this function runs _every_ time state or props change.
https://devtrium.com/posts/dependency-arrays